src/ECL/ECL.hs

```diff
@@ -31,6 +35,15 @@ data Binder lit
   | Record [(Name, Binder lit)]
   deriving (Show, Eq)
 
+-- | Represents constructors' arities.
+type Arity = Int
+
+-- | Environment
+--
+-- The language implementor should provide an environment to let
+-- the checker lookup for constructors' names (just for one type for now).
+type Environment = [(Name, Arity)]
```

Comment:
The first one makes the most sense to me. Let's not introduce `Type` if we don't need it.